src/app/backend/handler/apihandler.go

```diff
@@ -1412,19 +1412,31 @@ func (apiHandler *APIHandler) handlePutResource(
 
 func (apiHandler *APIHandler) handleDeleteResource(
 	request *restful.Request, response *restful.Response) {
-	verber, err := apiHandler.cManager.VerberClient(request)
-	if err != nil {
-		handleInternalError(response, err)
-		return
-	}
-
 	kind := request.PathParameter("kind")
 	namespace, ok := request.PathParameters()["namespace"]
 	name := request.PathParameter("name")
 
-	if err := verber.Delete(kind, ok, namespace, name); err != nil {
-		handleInternalError(response, err)
-		return
+	switch kind {
+	case api.ResourceKindJob:
+		k8sClient, err := apiHandler.cManager.Client(request)
+		if err != nil {
+			handleInternalError(response, err)
+			return
+		}
+		if err := job.DeleteJob(k8sClient, namespace, name); err != nil {
```

Comment:
If you want to keep client creation in the switch then at least move this error handling out of switch. Just assign error and handle it outside the switch. 

```golang
var err error
switch kind {
    case api.ResourceKindJob:
        k8sClient, err := apiHandler.cManager.Client(request)
        if err != nil {
            handleInternalError(response, err)
            return
        }

        err = job.DeleteJob(k8sClient, namespace, name)
    default:
        verber, err := apiHandler.cManager.VerberClient(request)
        if err != nil {
            handleInternalError(response, err)
            return
        }

        err = verber.Delete(kind, ok, namespace, name)
    }
}

if err != nil {
    handleInternalError(response, err)
    return
}
```